backend/tasks/user.py

```diff
@@ -0,0 +1,87 @@
+from flask_restful import Resource, reqparse
+from flask import request
+from psycopg2.extras import RealDictCursor
+from pymysql.cursors import DictCursor
+import os
+from flask import make_response
+import json
+import logging
+from flask_login import login_required
+logger = logging.getLogger(__name__)
+
+
+class UserTask(Resource):
+
+    @login_required
+    def get(self, user):
+        if user is None or len(user) == 0:
+            return make_response(json.dumps({'message': 'Parameter \'user\' is required'}), 400)
+
+        try:
+            with request.db.cursor(cursor_factory=RealDictCursor) if os.environ.get('IS_LOCAL') else request.db.cursor(DictCursor) as cur:
+                cur.execute("SELECT \"user\".username, \"user\".enabled FROM \"user\" WHERE \"user\".username=%s", (user,))
+                result = cur.fetchone()
+                if cur.rowcount == 0:
+                    return make_response(json.dumps({'message': f"Could not find user {user}"}), 404)
+                request.db.commit()
+                return make_response(json.dumps(result), 200)
+        except Exception as e:
+            logger.exception("Unable to fetch users: " + str(e))
+            return make_response(json.dumps({'message': f"Unable to fetch user {user}"}), 500)
+
+    # Used for enabling a user, under the general put function
+    # Trying to keep this as RESTful as possible, but may want to separate out into it's own 'enable' endpoint
+    @login_required
+    def put(self, user):
+        # Define params
+        parser = reqparse.RequestParser()
+        parser.add_argument('enable', type=str, location='args')
+        parser.add_argument('queue', type=str, location='args')
+        args = parser.parse_args()
+        enable = args.get('enable')
+        queue = args.get('queue')
+
+        # Check params
+        if enable is None or user is None:
+            return make_response(json.dumps({'message': 'Parameter \'enable\' required'}), 400)
```

Comment:
It can be "enable" or "user" here. So, message can be: 'Parameter \'enable\' and \'user\' required'?

Reply:
The user parameter is a part of the endpoint, so the only thing to check is enable. If you don't provide a user parameter you'll never actually hit this endpoint, so the user check here is useless but i think it can left in for now.

Reply:
Got it, thanks for the clarification.